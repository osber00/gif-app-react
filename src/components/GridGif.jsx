import PropTypes from 'prop-types'
import { GifItem } from "./GifItem";
import {useFetchGif} from '../hooks/useFetchGif'

export const GridGif = ({categoria}) => {
  const { gifs, isLoading } = useFetchGif(categoria);

  return (
    <div>
      <h4>{categoria}</h4>
      {isLoading && <h3>Cargando...</h3>}
      <div className='card-grid'>
        {gifs.map((gif) => (
          <GifItem key={gif.id} {...gif} />
        ))}
      </div>
    </div>
  );
}

GridGif.propTypes = {
  categoria : PropTypes.string.isRequired
}
