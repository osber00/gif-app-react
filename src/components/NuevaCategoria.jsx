import { useState } from "react";
import PropTypes from 'prop-types'

const NuevaCategoria = ({ onGuardarCategoria }) => {
  const [inputValue, setInputValue] = useState("");

  const onSubmit = (e) => {
    e.preventDefault();
    const categoria = inputValue.trim();
    if (categoria.length <= 1) return;
    onGuardarCategoria(categoria);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type='text'
        placeholder='Nueva categorÃ­a'
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
      />
    </form>
  );
};

NuevaCategoria.propTypes = {
  onGuardarCategoria: PropTypes.func.isRequired
}

export default NuevaCategoria;
