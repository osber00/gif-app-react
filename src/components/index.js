export {default as GifItem} from './GifItem'
export {default as GridGif} from './GridGif'
export {default as NuevaCategoria} from './NuevaCategoria'