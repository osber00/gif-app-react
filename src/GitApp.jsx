import { useState } from "react";
import {NuevaCategoria, GridGif} from './components'


export const GitApp = () => {
  const [categorias, setCategorias] = useState(['One punch']);

  const onGuardarCategoria = (nuevaCategoria) => {
    if (categorias.includes(nuevaCategoria)) return;
    setCategorias([nuevaCategoria, ...categorias]);
  };

  return (
    <>
      <h3>Categor√≠as</h3>
      <NuevaCategoria onGuardarCategoria={onGuardarCategoria} />
      <ul>
        {categorias.map((item) => (
          <GridGif key={item} categoria={item}/>
        ))}
      </ul>
    </>
  );
}
