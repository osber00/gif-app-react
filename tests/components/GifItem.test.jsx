import { render } from "@testing-library/react"
import { GifItem } from "../../src/components"

describe('Test del componente <GifItem>', () => {

    const titulo = 'Naturo'
    const url = 'https://hello/naruto.jpg'

    test('Match con el Snapshot', () => {
        const {container} = render(<GifItem title={titulo} url={url} />)
        expect(container).toMatchSnapshot()
    })
})