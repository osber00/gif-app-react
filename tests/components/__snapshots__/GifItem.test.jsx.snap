// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente <GifItem> Match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naturo"
      src="https://hello/naruto.jpg"
    />
    <p>
      Naturo
    </p>
  </div>
</div>
`;
